src/snowflake/connector/aio/_build_upload_agent.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+from io import BytesIO
+from logging import getLogger
+from typing import TYPE_CHECKING, cast
+
+from snowflake.connector import Error
+from snowflake.connector.bind_upload_agent import BindUploadAgent as BindUploadAgentSync
+from snowflake.connector.errors import BindUploadError
+
+if TYPE_CHECKING:
+    from snowflake.connector.aio import SnowflakeCursor
+
+logger = getLogger(__name__)
+
+
+class BindUploadAgent(BindUploadAgentSync):
+    def __init__(
+        self,
+        cursor: SnowflakeCursor,
+        rows: list[bytes],
+        stream_buffer_size: int = 1024 * 1024 * 10,
+    ) -> None:
+        super().__init__(cursor, rows, stream_buffer_size)
+        self.cursor = cast("SnowflakeCursor", cursor)
+
+    async def _create_stage(self) -> None:
+        await self.cursor.execute(self._CREATE_STAGE_STMT)
```

Comment:
should we log a jira to use scoped temp object since we will add it on sync connector?

Reply:
yes, I have a JIRA to track syncing all main branches into our aio connector: [intermediate](https://snowflakecomputing.atlassian.net/browse/SNOW-1664063)